From a2498c7aa29a155fbf707bbcb07580353979329c Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 01 Jul 2010 18:08:50 +0000
Subject: Don't show applet until after connecting to realize

Otherwise, the realize signals may run before the handler
is hooked up, causing crashes down the line.

https://bugzilla.gnome.org/show_bug.cgi?id=472773
---
diff --git a/applets/wncklet/workspace-switcher.c b/applets/wncklet/workspace-switcher.c
index 8e2a08e..02d18a8 100644
--- a/applets/wncklet/workspace-switcher.c
+++ b/applets/wncklet/workspace-switcher.c
@@ -546,8 +546,6 @@ workspace_switcher_applet_fill (PanelApplet *applet)
 	 * environments. See wnck_pager_set_orientation() doc */
 	pager_update (pager);
 
-	gtk_widget_show (pager->applet);
-
 	g_signal_connect (G_OBJECT (pager->applet),
 			  "realize",
 			  G_CALLBACK (applet_realized),
@@ -569,6 +567,8 @@ workspace_switcher_applet_fill (PanelApplet *applet)
 			  G_CALLBACK (applet_change_background),
 			  pager);
 
+	gtk_widget_show (pager->applet);
+
 	panel_applet_set_background_widget (PANEL_APPLET (pager->applet),
 					    GTK_WIDGET (pager->applet));
 	
--
cgit v0.8.3.1

